src/test/java/com/checkmarx/flow/cucumber/component/commentscript/CommentScriptSteps.java

```diff
@@ -0,0 +1,151 @@
+import com.checkmarx.flow.utils.ScanUtils;
+import com.checkmarx.sdk.config.CxProperties;
+import com.checkmarx.sdk.dto.ScanResults;
+import com.checkmarx.sdk.dto.cx.CxScanParams;
+import com.checkmarx.sdk.exception.CheckmarxException;
+import com.checkmarx.sdk.service.*;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import io.cucumber.spring.CucumberContextConfiguration;
+import lombok.extern.slf4j.Slf4j;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest(classes = {CxFlowApplication.class})
+@CucumberContextConfiguration
+@Slf4j
+public class CommentScriptSteps {
+
+    private final static String MEANINGLESS_STRING = "meaningless-string";
+    private final static String TOKEN = "Token";
+    private final static  String EMPTY_STRING = "";
+    private String branchName;
+    private String commentMessageFromRequest;
+    private CxClient cxClientMock;
+    ScanRequestConverter scanRequestConverterMock;
+    SastScanner sastScanner;
+    CxProperties cxProperties;
+    private FlowController flowController;
+    private FlowProperties flowProperties;
+
+    public CommentScriptSteps(FlowProperties flowProperties, CxProperties cxProperties, HelperService helperService,
+                              JiraProperties jiraProperties, FilterFactory filterFactory, ConfigurationOverrider configOverrider,
+                              ResultsService resultService, ProjectNameGenerator projectNameGenerator, BugTrackerEventTrigger btet){
+
+
+        cxClientMock = mock(CxClient.class);
+        scanRequestConverterMock = mock(ScanRequestConverter.class);
+
+        this.flowProperties = flowProperties;
+
+        sastScanner = mock(SastScanner.class, Mockito.withSettings().useConstructor(resultService, cxClientMock,
+                helperService, cxProperties, flowProperties, null, null, scanRequestConverterMock, btet, null));
+
+        this.cxProperties = cxProperties;
+        FlowService flowService = new FlowService(Collections.singletonList(sastScanner), projectNameGenerator, resultService);
+
+        this.flowController = new FlowController(flowProperties, cxProperties, flowService, helperService, jiraProperties, filterFactory, configOverrider, sastScanner);
+    }
+
+
+    @Before("@DeleteBranchFeature")
+    public void setMockers() throws CheckmarxException {
+        CreateScanAnswerer createScanAnswerer = new CreateScanAnswerer();
+        when(cxClientMock.getScanIdOfExistingScanIfExists(anyInt())).thenReturn(-1);
+        when(cxClientMock.createScan(any(), anyString())).thenAnswer(createScanAnswerer);
```

Comment:
changed to lambda instead of answerer class

only create scan and get-project methods are mocked.
the test is actually testing only the 'comment text' and not the scan flow